bpf-gpl/conntrack.h

```diff
@@ -200,7 +208,40 @@ static CALI_BPF_INLINE int calico_ct_v4_create_tracking(struct ct_create_ctx *ct
 		CALI_DEBUG("CT-ALL Whitelisted dest side - to EP\n");
 	}
 
-	err = cali_v4_ct_update_elem(k, &ct_value, 0);
+	err = cali_v4_ct_update_elem(k, &ct_value, BPF_NOEXIST);
+
+	if (err == -17 /* EEXIST */) {
+		CALI_DEBUG("Source collision for 0x%x:%d\n", bpf_htonl(ip_src), sport);
```

Comment:
It is possible, that there is a connection with the same tuple that is already expired but not collected by the cleaner yet. We do not treat expiration in BPF yet. It is racy with the cleaner. Nevertheless, it is a known issue that is tcp_tw_reuse is set, we can hit this problem (tracked as a separate issue). Afaict this would also (partially) resolve that tcp_tw_reuse issue as it would pick a different port. Not sure if that is the best resolution. We may need to do the expiration in BPF and be more clever about the race.